### PR TITLE
Replace deprecated pkg_resources lib with importlib_resources

### LINKED ISSUE #99: Deprecated pkg_resources package warning
```
../../../../../usr/share/miniconda/envs/__setup_conda/lib/python3.11/site-packages/mongomock/__version__.py:1
  /usr/share/miniconda/envs/__setup_conda/lib/python3.11/site-packages/mongomock/__version__.py:1: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
    import pkg_resources
```

### PR DESCRIPTION
**Pull request by northwestwitch**
Fix #99 

**How to test**:
1. All automatic tests should run successfully

**Review:**
- [x] code approved by HS
- [x] tests executed by GitHub actions

This is a patch version PR

### CODE REVIEW

File: mutacc/resources/__init__.py [hunk 1]
```diff
@@ -1,27 +1,22 @@
...
+from importlib_resources import files
 
-case_filename = "resources/case.yaml"
-father_fastq1_filename = "resources/father_R1.fastq.gz"
-father_fastq2_filename = "resources/father_R2.fastq.gz"
-father_bam_filename = "resources/father.bam"
-mother_bam_filename = "resources/mother.bam"
-child_bam_filename = "resources/child.bam"
-vcf_filename = "resources/variant1.vcf.gz"
+BASE_PATH = "mutacc.resources"
```
Comment (henrikstranneheim) [inline mutacc/resources/__init__.py:3]: Add type hints

File: mutacc/resources/__init__.py [hunk 2]
```diff
@@ -1,27 +1,22 @@
...
-)
-path_to_background_fastq2_file = pkg_resources.resource_filename(
-    "mutacc", father_fastq2_filename
-)
-path_to_background_bam_file = pkg_resources.resource_filename(
-    "mutacc", father_bam_filename
-)
 
-default_vcf_parser = pkg_resources.resource_filename("mutacc", vcf_parser_file)
+path_to_case_file = str(files(BASE_PATH).joinpath(case_filename))
```
Comment (henrikstranneheim) [inline mutacc/resources/__init__.py:15]: Use pathlib instead?

**Review by henrikstranneheim — APPROVED** [review]
👍

File: mutacc/resources/__init__.py (hunk 2 above)
Comment (northwestwitch) [reply mutacc/resources/__init__.py:15]: importlib.resources is a context manager for pathlib that basically gives you the pathlib.Path. 
I've followed the instructions given by the [pkg_resources migration guide] (https://importlib-resources.readthedocs.io/en/latest/migration.html). And we use importlib_resources anyway when retrieving the software version so why not also here?

File: mutacc/resources/__init__.py (hunk 1 above)
Comment (northwestwitch) [reply mutacc/resources/__init__.py:3]: Done

**Pull request merged**